src/components/budget-expiry-notification/data/hooks/useExpiry.jsx

```diff
@@ -0,0 +1,80 @@
+import { useState, useEffect } from 'react';
+import {
+  getEnterpriseBudgetExpiringAlertCookieName,
+  getEnterpriseBudgetExpiringModalCookieName,
+  getExpirationMetadata,
+} from '../utils';
+
+const useExpiry = (enterpriseId, budget, modalOpen, modalClose, alertOpen, alertClose) => {
+  const [alert, setAlert] = useState(null);
+  const [expirationThreshold, setExpirationThreshold] = useState(null);
+  const [modal, setModal] = useState(null);
+
+  useEffect(() => {
+    if (!budget || budget.length === 0) {
+      return;
+    }
+
+    const { thresholdKey, threshold } = getExpirationMetadata(budget.end);
+
+    if (thresholdKey !== null) {
+      const { alertTemplate, modalTemplate } = threshold;
+
+      setAlert(alertTemplate);
+      setModal(modalTemplate);
+      setExpirationThreshold({
+        thresholdKey,
+        threshold,
+      });
+    }
+
+    const seenCurrentExpiringModalCookieName = getEnterpriseBudgetExpiringModalCookieName({
+      expirationThreshold: thresholdKey,
+      enterpriseId,
+    });
+
+    const seenCurrentExpiringAlertCookieName = getEnterpriseBudgetExpiringAlertCookieName({
+      expirationThreshold: thresholdKey,
+      enterpriseId,
+    });
+
+    const isModalDismissed = global.localStorage.getItem(seenCurrentExpiringModalCookieName);
+    const isAlertDismissed = global.localStorage.getItem(seenCurrentExpiringAlertCookieName);
+
+    if (!isModalDismissed) {
+      modalOpen();
+    }
+
+    if (!isAlertDismissed) {
+      alertOpen();
+    }
+  }, [budget, modalOpen, alertOpen, enterpriseId]);
+
+  const dismissModal = () => {
+    const seenCurrentExpirationModalCookieName = getEnterpriseBudgetExpiringModalCookieName({
+      expirationThreshold: expirationThreshold.thresholdKey,
+      enterpriseId,
+    });
+
+    global.localStorage.setItem(seenCurrentExpirationModalCookieName, 'true');
+
+    modalClose();
+  };
+
+  const dismissAlert = () => {
+    const seenCurrentExpirationAlertCookieName = getEnterpriseBudgetExpiringAlertCookieName({
+      expirationThreshold: expirationThreshold.thresholdKey,
+      enterpriseId,
+    });
+
+    global.localStorage.setItem(seenCurrentExpirationAlertCookieName, 'true');
+
+    alertClose();
+  };
```

Comment:
[curious] Wondering if its worth putting track events in here as well when the modal is dismissed?

Reply:
I'm not entirely convinced that we'll gather valuable metrics by tracking dismissals without any user identifier other than the enterpriseUUID. In cases where we track contact support or representative information, it can be extrapolated to an organization to take action if we receive thousands of support contacts, indicating a potential issue. But I'd like to know your thoughts on this?